### PR TITLE
Make notation for Pr consistent

### LINKED ISSUE #602: Use `\Pr` for probability in docs
#### Summary:

LaTeX has a built-in `\Pr` that renders as $\Pr$.  We should replace all of our hand-built versions, including
`\operatorname{Pr}`, `\textsf{Pr}`, `\mbox{Pr}`, and `\textrm{Pr}`.

Thanks to @dvukcevic for pointing this out in comment https://github.com/stan-dev/docs/pull/373#issuecomment-1341677732.

#### Description:

See summary.

#### Current Version:

v2.18.0

### PR DESCRIPTION
**Pull request by WardBrian**
#### Submission Checklist

- [x] Builds locally 
- [x] New functions marked with `` `r since("VERSION")` ``
- [x] Declare copyright holder and open-source license: see below

#### Summary

Closes #602

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC BY-ND 4.0 (https://creativecommons.org/licenses/by-nd/4.0/)

### CODE REVIEW

File: src/stan-users-guide/finite-mixtures.Rmd [hunk 1]
```diff
@@ -162,7 +162,7 @@ The posterior $p(z_n \mid y_n, \mu, \sigma)$ over the mixture indicator $z_n
 posterior probability that that observation $y_n$ was generated by
 mixture component $k$.  The posterior can be computed via Bayes's rule,
 \begin{align*}
-\operatorname{Pr}\left(z_n = k \mid y_n, \mu, \sigma, \lambda \right)
+\Pr\left(z_n = k \mid y_n, \mu, \sigma, \lambda \right)
```
Comment (bob-carpenter) [inline src/stan-users-guide/finite-mixtures.Rmd:165]: '(' to '[' and similarly for closing parens.  If you are using `\left[`, then it should have a bit of negative space, `\Pr\!\left[` in order to not leave too much space between the "Pr" and the "[".   The alternative is to make \Pr take an argument and put it in the brackets, i.e.,

```latex
\newcommand{\Pr}[1]{\textrm{Pr}\!\left[ #1 \right]}
```

File: src/stan-users-guide/finite-mixtures.Rmd [hunk 2]
```diff
@@ -179,8 +179,8 @@ $$
 
 On the log scale, the normalized probability is computed as
 \begin{align*}
-\log\,&\operatorname{Pr}\left(z_n = k \mid y_n, \mu, \sigma, \lambda\right) \\
- &= \log p\left(y_n \mid z_n = k, \mu, \sigma\right) + \log \operatorname{Pr}\left(z_n = k \mid \lambda\right) \\
+\log\,&\Pr\left(z_n = k \mid y_n, \mu, \sigma, \lambda\right) \\
```
Comment (bob-carpenter) [inline src/stan-users-guide/finite-mixtures.Rmd:182]: round to square

**Review by bob-carpenter — CHANGES_REQUESTED** [review]
Two things.

1.  The definition of `\Pr` should be `\textrm{Pr}`.  If not, please make it that.
2.  We probably want to go with the `Pr[...]` form with square brackets.<sup>*</sup>  So I marked the one with parens that I saw.

<hr />
<sup>*</sup> The convention for round vs. square brackets (imported from physics, I believe) is that higher-order functions that apply to other functions use square brackets.  That's why E[] and Pr[] use square brackets, but p() uses round brackets---the former take random variables as arguments (which are functions), whereas p() just takes primitive arguments.

**Comment by WardBrian** [comment]
The original issue was opened because `\Pr` is built in: $\Pr$ 

It looks identical to `\mathrm{Pr}` in practice: $\mathrm{Pr}$

**Comment by WardBrian** [comment]
@bob-carpenter mind giving this another look?

**Review by bob-carpenter — APPROVED** [review]
Thanks.  Looks good.

**Pull request merged**